src/workerd/api/tests/crypto-extras-test.js

```diff
@@ -0,0 +1,139 @@
+import {
+  strictEqual,
+  ok,
+  throws
+} from 'node:assert';
+
+export const timingSafeEqual = {
+  test() {
+
+    // Note that this does not actually test that the equality check is,
+    // in fact, timing safe. It checks only the basic operation of the API
+
+    const enc = new TextEncoder();
+    [
+      [new ArrayBuffer(0), new ArrayBuffer(0)],
+      [new ArrayBuffer(1), new ArrayBuffer(1)],
+      [enc.encode('hello'), enc.encode('hello')],
+      [enc.encode('hellothere'), enc.encode('hellothere').buffer],
+    ].forEach(([a, b]) => {
+      if (!crypto.subtle.timingSafeEqual(a, b)) {
+        throw new Error('inputs should have been equal', a, b);
+      }
+    });
+
+    [
+      [enc.encode('hello'), enc.encode('there')],
+      [new Uint8Array([1,2,3,4]), new Uint32Array([1])],
+    ].forEach(([a, b]) => {
+      if (crypto.subtle.timingSafeEqual(a, b)) {
+        throw new Error('inputs should not have been equal', a, b);
+      }
+    });
+
+    [
+      ['hello', 'there'],
+      [new ArrayBuffer(0), new ArrayBuffer(1)],
+    ].forEach(([a, b]) => {
+      try {
+        crypto.subtle.timingSafeEqual(a, b);
+        throw new Error('inputs should have caused an error', a, b);
+      } catch {}
+    });
+
+  }
+};
+
+export const randomUuid = {
+  test() {
+    const pattern = /[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[ab89][a-f0-9]{3}-[a-f0-9]{12}/;
+    // Loop through a bunch of generated UUID's to make sure we're consistently successful.
+    for (let n = 0; n < 100; n++) {
+      const uuid = crypto.randomUUID();
+      if (!pattern.test(uuid)) {
+        throw new Error(`${uuid} is not a valid random UUID`);
+      }
+    }
```

Comment:
A "no repeats" check would be good here too.

Reply:
will add that separately